sdk/keyvault/keyvault-secrets/review/keyvault-secrets.review.api.md

```diff
@@ -0,0 +1,151 @@
+export interface NewPipelineOptions {
+    // (undocumented)
+    HTTPClient?: HttpClient;
+    // (undocumented)
+    logger?: HttpPipelineLogger;
+    // (undocumented)
+    proxyOptions?: ProxyOptions;
+    // (undocumented)
+    retryOptions?: RetryOptions;
+    telemetry?: TelemetryOptions;
+}
+
+export { PagedAsyncIterableIterator }
+
+export { PageSettings }
+
+// @public (undocumented)
+export interface ParsedKeyVaultEntityIdentifier {
+    name: string;
+    vaultUrl: string;
+    version?: string;
+}
+
+// @public
+export interface ProxyOptions {
+    // (undocumented)
+    proxySettings?: string;
+}
+
+// @public
+export interface RetryOptions {
+    readonly maxRetryDelayInMs?: number;
+    readonly retryCount?: number;
+    readonly retryIntervalInMS?: number;
+}
+
+// @public
+export interface Secret {
+    properties: SecretProperties;
+    value?: string;
+}
+
+// @public
+export interface SecretProperties extends ParsedKeyVaultEntityIdentifier {
+    contentType?: string;
+    readonly created?: Date;
+    enabled?: boolean;
+    expires?: Date;
+    id?: string;
+    readonly keyId?: string;
+    readonly managed?: boolean;
+    notBefore?: Date;
+    readonly recoveryLevel?: DeletionRecoveryLevel;
+    tags?: {
+        [propertyName: string]: string;
+    };
+    readonly updated?: Date;
+}
+
+// @public
+export class SecretsClient {
+    constructor(url: string, credential: TokenCredential, pipelineOrOptions?: ServiceClientOptions | NewPipelineOptions);
+    backupSecret(secretName: string, options?: RequestOptionsBase): Promise<Uint8Array | undefined>;
+    protected readonly credential: TokenCredential;
+    deleteSecret(secretName: string, options?: RequestOptionsBase): Promise<DeletedSecret>;
+    static getDefaultPipeline(credential: TokenCredential, pipelineOptions?: NewPipelineOptions): ServiceClientOptions;
+    getDeletedSecret(secretName: string, options?: RequestOptionsBase): Promise<DeletedSecret>;
+    getSecret(secretName: string, options?: GetSecretOptions): Promise<Secret>;
+    listDeletedSecrets(options?: GetSecretsOptions): PagedAsyncIterableIterator<SecretProperties, SecretProperties[]>;
+    listSecrets(options?: GetSecretsOptions): PagedAsyncIterableIterator<SecretProperties, SecretProperties[]>;
+    listSecretVersions(secretName: string, options?: GetSecretsOptions): PagedAsyncIterableIterator<SecretProperties, SecretProperties[]>;
+    readonly pipeline: ServiceClientOptions;
+    purgeDeletedSecret(secretName: string, options?: RequestOptionsBase): Promise<void>;
+    recoverDeletedSecret(secretName: string, options?: RequestOptionsBase): Promise<Secret>;
+    restoreSecret(secretBundleBackup: Uint8Array, options?: RequestOptionsBase): Promise<Secret>;
+    setSecret(secretName: string, value: string, options?: SetSecretOptions): Promise<Secret>;
+    updateSecretProperties(secretName: string, secretVersion: string, options?: UpdateSecretOptions): Promise<Secret>;
+    readonly vaultBaseUrl: string;
```

Comment:
vaultUrl over vaultBaseUrl (as per discussion in .net review)

Reply:
Agreed.

Reply:
Or if we change to endpoint we should change this universal throughout the libraries.